pkg/cableengine/cableengine.go

```diff
@@ -1,11 +1,118 @@
 package cableengine
 
 import (
+	"fmt"
+	"reflect"
+	"sync"
+
+	"github.com/submariner-io/submariner/pkg/cable"
+	"github.com/submariner-io/submariner/pkg/log"
 	"github.com/submariner-io/submariner/pkg/types"
+	"github.com/submariner-io/submariner/pkg/util"
+	"k8s.io/klog"
+
+	// Add supported drivers
+	_ "github.com/submariner-io/submariner/pkg/cable/ipsec"
 )
 
+// Engine represents an implementation of some remote connectivity mechanism, such as
+// a VPN gateway.
+// An Engine cooperates with, and delegates work to, a cable.Driver for implementing
+// a secure connection to remote clusters.
 type Engine interface {
+	// StartEngine performs any general set up work needed independent of any remote connections.
 	StartEngine() error
-	InstallCable(types.SubmarinerEndpoint) error
-	RemoveCable(types.SubmarinerEndpoint) error
+	// InstallCable performs any set up work needed for connecting to given remote endpoint.
+	// Once InstallCable completes, it should be possible to connect to remote
+	// Pods or Services behind the given endpoint.
+	InstallCable(remote types.SubmarinerEndpoint) error
+	// RemoveCable disconnects the Engine from the given remote endpoint. Upon completion.
+	// remote Pods and Service may not be accessible any more.
+	RemoveCable(remote types.SubmarinerEndpoint) error
+}
+
+type engine struct {
+	sync.Mutex
+
+	driver        cable.Driver
+	localSubnets  []string
+	localCluster  types.SubmarinerCluster
+	localEndpoint types.SubmarinerEndpoint
+}
+
+// NewEngine creates a new Engine for the local cluster
+func NewEngine(localSubnets []string, localCluster types.SubmarinerCluster, localEndpoint types.SubmarinerEndpoint) (Engine, error) {
+	driver, err := cable.NewDriver(localSubnets, localEndpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	return &engine{
+		localCluster:  localCluster,
+		localEndpoint: localEndpoint,
+		localSubnets:  localSubnets,
+		driver:        driver,
+	}, nil
+}
+
+func (i *engine) StartEngine() error {
+	klog.Infof("Starting IPSec Engine (Charon)")
```

Comment:
This is IPSEC/Strongswan specific, although, just a nit, something we can fix on a follow up patch.

Reply:
you're right - I'll push a follow-up for this.